dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/QueryModifiers.java

```diff
@@ -75,4 +75,53 @@
      */
     @JsonProperty
     private final Date maxDate;
+
+    /**
+     * The sub-expression used to query the analytics value column.
+     */
+    @JsonProperty
+    private final String subExpression;
+
+    /**
+     * The value type of the sub-expression (can be different from the value
+     * type of the object).
+     */
+    @JsonProperty
+    private final ValueType valueType;
+
+    // -------------------------------------------------------------------------
+    // Logic
+    // -------------------------------------------------------------------------
+
+    /**
+     * Gets the query modifiers that matter for analytics grouping.
+     */
+    public QueryModifiers getQueryModsForAnalyticsGrouping()
+    {
+        return this.toBuilder().periodOffset( 0 ).build();
+    }
+
+    /**
+     * Gets an Id for these query modifiers that matter for analytics grouping.
+     */
+    public String getQueryModsId()
+    {
+        QueryModifiers analyticsQueryMods = getQueryModsForAnalyticsGrouping();
+
+        return analyticsQueryMods.isDefault()
+            ? null
+            : Integer.toHexString( analyticsQueryMods.hashCode() );
+    }
+
+    // -------------------------------------------------------------------------
+    // Supportive methods
+    // -------------------------------------------------------------------------
+
+    /**
+     * Checks if these query mods have all default values.
+     */
+    private boolean isDefault()
+    {
+        return this.equals( QueryModifiers.builder().build() );
```

Comment:
Maybe there should be a `DEFAULT` constant for `QueryModifiers.builder().build()` then in this class.

Reply:
Good idea, done.